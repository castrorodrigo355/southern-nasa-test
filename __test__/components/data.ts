import {
  CameraName,
  FullName,
  Photo,
  RoverName,
  Status,
} from '@/types/photo.interface';

export const data: Photo[] = [
  {
    id: 102693,
    sol: 1000,
    camera: {
      id: 20,
      name: 'FHAZ' as CameraName,
      rover_id: 5,
      full_name:
        'Front Hazard Avoidance Camera' as FullName.RearHazardAvoidanceCamera,
    },
    img_src:
      'http://mars.jpl.nasa.gov/msl-raw-images/proj/msl/redops/ods/surface/sol/01000/opgs/edr/fcam/FLB_486265257EDR_F0481570FHAZ00323M_.JPG',
    earth_date: new Date('2015-05-30'),
    rover: {
      id: 5,
      name: 'Curiosity' as RoverName,
      landing_date: new Date('2012-08-06'),
      launch_date: new Date('2011-11-26'),
      status: 'active' as Status,
    },
  },
  {
    id: 424939,
    sol: 1000,
    camera: {
      id: 22,
      name: 'MAST' as CameraName,
      rover_id: 5,
      full_name: 'Mast Camera' as FullName.MastCamera,
    },
    img_src:
      'http://mars.jpl.nasa.gov/msl-raw-images/msss/01000/mcam/1000MR0044631130503673E01_DXXX.jpg',
    earth_date: new Date('2015-05-30'),
    rover: {
      id: 5,
      name: 'Curiosity' as RoverName,
      landing_date: new Date('2012-08-06'),
      launch_date: new Date('2011-11-26'),
      status: 'active' as Status,
    },
  },
];
