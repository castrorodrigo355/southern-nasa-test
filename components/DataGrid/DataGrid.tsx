import { isFavorite } from '@/helpers';
import { useLocalStorage } from '@/hooks/useLocalStorage';
import { Photo } from '@/types/photo.interface';
import { Grid } from '@chakra-ui/react';
import { GridItem as GridItemElement } from '../GridItem';

interface DataGridProps {
  data: Photo[];
}

export const DataGrid = ({ data }: DataGridProps): JSX.Element => {
  const { favorites, onClickFavorite } = useLocalStorage<number>(
    'favorites',
    []
  );

  return (
    <Grid display='flex' flexWrap='wrap' placeContent='center' gap={6}>
      {data.map((item) => (
        <GridItemElement
          key={item.id}
          item={item}
          favorite={isFavorite(favorites, item.id)}
          onClickFavorite={() => onClickFavorite(item.id)}
        />
      ))}
    </Grid>
  );
};
