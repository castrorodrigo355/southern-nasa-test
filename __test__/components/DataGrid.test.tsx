import { render, screen } from '@testing-library/react';
import { DataGrid } from '../../components/DataGrid';
import { data } from './data';

describe('<DataGrid />', () => {
  test('1. Snapshot: ', () => {
    const { container } = render(<DataGrid data={data} />);
    expect(container).toMatchSnapshot();
  });

  test('2. Should display heading: ', () => {
    render(<DataGrid data={data} />);
    const images = screen.getAllByRole('img');
    expect(images).toBeTruthy();
    expect(images.length).toBeGreaterThan(0);
  });
});
