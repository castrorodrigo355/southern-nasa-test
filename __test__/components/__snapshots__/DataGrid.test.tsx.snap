// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DataGrid /> 1. Snapshot:  1`] = `
<div>
  <div
    class="css-17x2qeu"
  >
    <div
      class="chakra-card css-136b6hl"
    >
      <div
        class="chakra-card__body css-0"
      >
        <div
          class="chakra-stack css-1qe7oq1"
        >
          <img
            alt="..."
            data-nimg="1"
            decoding="async"
            fetchpriority="high"
            height="300"
            src="/_next/image?url=http%3A%2F%2Fmars.jpl.nasa.gov%2Fmsl-raw-images%2Fproj%2Fmsl%2Fredops%2Fods%2Fsurface%2Fsol%2F01000%2Fopgs%2Fedr%2Ffcam%2FFLB_486265257EDR_F0481570FHAZ00323M_.JPG&w=640&q=75"
            srcset="/_next/image?url=http%3A%2F%2Fmars.jpl.nasa.gov%2Fmsl-raw-images%2Fproj%2Fmsl%2Fredops%2Fods%2Fsurface%2Fsol%2F01000%2Fopgs%2Fedr%2Ffcam%2FFLB_486265257EDR_F0481570FHAZ00323M_.JPG&w=384&q=75 1x, /_next/image?url=http%3A%2F%2Fmars.jpl.nasa.gov%2Fmsl-raw-images%2Fproj%2Fmsl%2Fredops%2Fods%2Fsurface%2Fsol%2F01000%2Fopgs%2Fedr%2Ffcam%2FFLB_486265257EDR_F0481570FHAZ00323M_.JPG&w=640&q=75 2x"
            style="color: transparent;"
            width="300"
          />
          <div
            class="chakra-stack css-e972zz"
          >
            <h2
              class="chakra-heading css-0"
            >
              Front Hazard Avoidance Camera
            </h2>
            <p
              class="chakra-text css-0"
            >
              Sol: 
              1000
            </p>
            <p
              class="chakra-text css-0"
            >
              Earth Date: 
              5/29/2015, 9:00:00 PM
            </p>
          </div>
        </div>
      </div>
      <hr
        aria-orientation="horizontal"
        class="chakra-divider css-1upb9tn"
      />
      <div
        class="chakra-card__footer css-zeh8nu"
      >
        <p
          class="chakra-text css-1ppa63l"
        >
          Camera: 
          FHAZ
        </p>
        <div
          class="chakra-button__group css-ejjise"
          data-orientation="horizontal"
          role="group"
        >
          <div
            class="chakra-stack css-x4g4hj"
          >
            <div
              class="chakra-offset-slide"
              style="opacity: 0; transform: translateX(-250px) translateY(8px) translateZ(0);"
            >
              <svg
                class="chakra-icon css-lsar2u"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </div>
          <div
            class="chakra-stack css-x4g4hj"
          >
            <div
              class="chakra-offset-slide"
              style="opacity: 0; transform: translateX(250px) translateY(8px) translateZ(0);"
            >
              <svg
                class="chakra-icon css-m5iyps"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="chakra-card css-136b6hl"
    >
      <div
        class="chakra-card__body css-0"
      >
        <div
          class="chakra-stack css-1qe7oq1"
        >
          <img
            alt="..."
            data-nimg="1"
            decoding="async"
            fetchpriority="high"
            height="300"
            src="/_next/image?url=http%3A%2F%2Fmars.jpl.nasa.gov%2Fmsl-raw-images%2Fmsss%2F01000%2Fmcam%2F1000MR0044631130503673E01_DXXX.jpg&w=640&q=75"
            srcset="/_next/image?url=http%3A%2F%2Fmars.jpl.nasa.gov%2Fmsl-raw-images%2Fmsss%2F01000%2Fmcam%2F1000MR0044631130503673E01_DXXX.jpg&w=384&q=75 1x, /_next/image?url=http%3A%2F%2Fmars.jpl.nasa.gov%2Fmsl-raw-images%2Fmsss%2F01000%2Fmcam%2F1000MR0044631130503673E01_DXXX.jpg&w=640&q=75 2x"
            style="color: transparent;"
            width="300"
          />
          <div
            class="chakra-stack css-e972zz"
          >
            <h2
              class="chakra-heading css-0"
            >
              Mast Camera
            </h2>
            <p
              class="chakra-text css-0"
            >
              Sol: 
              1000
            </p>
            <p
              class="chakra-text css-0"
            >
              Earth Date: 
              5/29/2015, 9:00:00 PM
            </p>
          </div>
        </div>
      </div>
      <hr
        aria-orientation="horizontal"
        class="chakra-divider css-1upb9tn"
      />
      <div
        class="chakra-card__footer css-zeh8nu"
      >
        <p
          class="chakra-text css-1ppa63l"
        >
          Camera: 
          MAST
        </p>
        <div
          class="chakra-button__group css-ejjise"
          data-orientation="horizontal"
          role="group"
        >
          <div
            class="chakra-stack css-x4g4hj"
          >
            <div
              class="chakra-offset-slide"
              style="opacity: 0; transform: translateX(-250px) translateY(8px) translateZ(0);"
            >
              <svg
                class="chakra-icon css-lsar2u"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </div>
          <div
            class="chakra-stack css-x4g4hj"
          >
            <div
              class="chakra-offset-slide"
              style="opacity: 0; transform: translateX(250px) translateY(8px) translateZ(0);"
            >
              <svg
                class="chakra-icon css-m5iyps"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
